gh-skeleton

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Default organizations
+DEST_ORG="cisagov"
+SRC_ORG="cisagov"
+
+# Define terminal colors for use in logger functions
+BLUE="\x1B[34m"
+GREEN="\x1B[32m"
+RED="\x1B[31m"
+RESET="\x1B[0m"
+YELLOW="\x1B[33m"
+
+log_info() {
+  echo -e "${BLUE}* $*${RESET}"
+}
+
+log_ok() {
+  echo -e "${GREEN}✓ $*${RESET}"
+}
+
+log_warn() {
+  echo -e "${YELLOW}⚠︎ $*${RESET}"
+}
+
+log_error() {
+  echo -e "${RED}! $*${RESET}" >&2
+}
+
+# Determine if an executable is in the PATH
+if ! type -p perl > /dev/null; then
+  log_error "Perl not found on the system and is required"
+  exit 1
+fi
+
+USAGE=$(
+  cat << END_OF_LINE
+gh extension to start a new GitHub project from a skeleton GitHub repository.
+
+Usage:
+  gh skeleton (-h | --help)
+  gh skeleton list [--src-org <name>]
+  gh skeleton clone [options] <parent-repo-name> <new-repo-name>
+
+Options:
+  -c --change-dir <dir> Create clone in this directory.
+  -d --dest-org <name>  Organization to create clone into [default: ${DEST_ORG}].
+  -h --help             Show this message.
+  -s --src-org <name>   Organization to search for skeletons [default: ${SRC_ORG}].
+END_OF_LINE
+)
+
+print_available_skeletons() {
+  src_org=$1
+  QUERY="
+  query {
+    search(query: \"org:$src_org topic:skeleton archived:false\", type: REPOSITORY, first: 100) {
+      repositoryCount
+      edges {
+        node {
+          ... on Repository {
+            name
+            description
+          }
+        }
+      }
+    }
+  }
+  "
+
+  # shellcheck disable=2016
+  # Disable single quote expansion warning as we want to suppress
+  # parameter expansion in a Go template.
+  TEMPLATE='
+    {{- range $repo := .data.search.edges -}}
+      {{- printf "%-25s\t%s\n" $repo.node.name $repo.node.description -}}
+    {{- end -}}
+  '
+
+  echo "Available skeletons in $src_org:"
+  echo
+  exec gh api graphql \
+    --raw-field query="${QUERY}" --paginate \
+    --template="${TEMPLATE}" | sort
+}
+
+clone_repo() {
+  src_repo=$1
+  dest_repo=$2
+  src_org=$3
+  dest_org=$4
+
+  # Before we continue check to see if the user is logged in. If there is no
+  # currently authenticated user gh will ask the user to run the "gh auth login"
+  # command and exit with return code 4.  This will cause this script to halt.
+  username=$(gh api user --jq '.login')
+
+  # If we got here we know we have an authenticated user.
+  log_ok "You are logged into GitHub as ${username}."
+
+  log_info "Creating ${dest_org}/${dest_repo} from skeleton ${src_org}/${src_repo}."
+
+  log_info "Cloning skeleton remote repository to the new local repository."
+  git clone --origin "${src_repo}" git@github.com:"${src_org}/${src_repo}".git "${dest_repo}"
```

Comment:
Are you sure @mcdonnnj?  All my remotes are named `skeleton-docker`, `skeleton-ansible-role`, etc.

Reply:
I was referring to things like the [cisagov/action-lineage conflict PR description](https://github.com/cisagov/action-lineage/pull/30).

Relavant portion:

```console
git clone git@github.com:cisagov/action-lineage.git action-lineage
cd action-lineage
git remote add skeleton https://github.com/cisagov/skeleton-python-library.git
git remote set-url --push skeleton no_push
git switch develop
git checkout -b lineage/skeleton --track origin/develop
git pull skeleton HEAD
git checkout --ours -- .github/lineage.yml
git add .github/lineage.yml
git status
```

Reply:
I see that the [`skeleton` script](https://github.com/cisagov/development-guide/blob/d15d4445ee56610d6d4059192ee984aee9f037e2/project_setup/scripts/skeleton#L59-L62) mirrors what you're talking about. In that case we need to get consistent language in place everywhere we're directing things about setting up repository specifics with `git`.

Reply:
Again, `git remote add skeleton` is incorrect in my opinion.  I think we want the `${src_repo}` there as the name of the skeleton remote because it provides useful information.  If this is inconsistent with instructions given elsewhere then I suggest changing the other instructions.

Reply:
I agree.  Let's find the documentation that is in conflict, and update it.

Reply:
I've created https://github.com/cisagov/action-lineage/issues/36 to track fixing this in that repository. We probably want to consider getting this into [cisagov/development-guide](https://github.com/cisagov/development-guide) at some point to solidify the expectation.